modules/ROOT/pages/zero-to-hero.adoc

```diff
@@ -0,0 +1,294 @@
+can use the ec-cli image if you need one, `quay.io/enterprise-contract/ec-cli:latest`.
+
+Let's copy the image to our test repository. This can be done with a tool such as
+https://github.com/containers/skopeo[skopeo] or
+https://github.com/google/go-containerregistry/tree/main/cmd/crane[crane]:
+
+[.console-input]
+[source, bash]
+----
+crane copy quay.io/enterprise-contract/ec-cli:latest "$REPOSITORY:latest"
+# or with skopeo:
+# skopeo copy docker://quay.io/enterprise-contract/ec-cli:latest \
+#  "docker://$REPOSITORY:latest" --dest-tls-verify=false
+# NOTE: Only set `--dest-tls-verify=false` if you are using a local registry.
+----
+
+== Generate Signining Key
+
+In order to sign and attest the image, we need a signing key.
+
+[.console-input]
+[source, bash]
+----
+cosign generate-key-pair
+----
+
+Enter a new password for the key-pair on the password prompt. For the purpose of this guide, use
+something memorable like `top-secret`. Otherwise, always use a strong secret password.
+
+The command should generate two files in the currently working directory. `cosign.pub` is the public
+key. We will use this later during verification. This can, and should, be distributed freely to
+anyone that may want to verify the image. `cosign.key` is the private key. This is used for signing
+content. Treat this like the password you provided. No one other than the entity responsible for
+singing and attestating images should have access to this.
+
+== Signing the Image
+
+We use cosign to sign our image.
+
+[.console-input]
+[source, bash]
+----
+cosign sign --key cosign.key $REPOSITORY:latest
+----
+
+Enter the key-pair password and `y` to upload a transaction record to
+https://docs.sigstore.dev/logging/overview/[Rekor].
+
+== Attesting the Image
+
+Next, let's use cosign again to associate a SLSA Provenance attestion to the image.
+
+Usually, the build system responsible for building the container image is also responsible for
+providing the SLSA Provenance attestation. Here, we will use an example one instead.
+
+First, create the content of the SLSA Provenance:
+
+[.console-input]
+[source, bash]
+----
+echo '{
+  "builder": {
+    "id": "https://localhost/dummy-id"
+  },
+  "buildType": "https://localhost/dummy-type",
+  "invocation": {},
+  "buildConfig": {},
+  "metadata": {
+    "buildStartedOn": "2023-09-25T16:26:44Z",
+    "buildFinishedOn": "2023-09-25T16:28:59Z",
+    "completeness": {
+      "parameters": false,
+      "environment": false,
+      "materials": false
+    },
+    "reproducible": false
+  },
+  "materials": []
+}
+' > predicate.json
+----
+
+Now, create an attestation and attach it to the container image.
+
+[.console-input]
+[source, bash]
+----
+cosign attest --predicate predicate.json --type slsaprovenance --key cosign.key "$REPOSITORY:latest"
+----
+
+As done while signing the image, enter the key-pair password and `y` to upload a transaction record
```

Comment:
```suggestion
All done with signing the image, enter the key-pair password and `y` to upload a transaction record
```
Maybe?

Reply:
This edit doesn't make sense to me, but what I wrote is probably confusing. I meant to say here that now we do the same thing we did when we ran the cosign sign command.

Reply:
You could say something like in the "signing image step", but honestly now that you explained it it makes a lot more sense.

Reply:
Alright. I'll leave it as is then.